infra/Pos/Communication/Types/Protocol.hs

```diff
@@ -291,12 +297,15 @@ data MkListeners m = MkListeners
         -- ^ Aggregated specs for which outgoing connections we might initiate
         }
 
-instance Monad m => Monoid (MkListeners m) where
-    mempty = MkListeners (\_ _ -> []) mempty mempty
-    a `mappend` b = MkListeners act (inSpecs a `mappend` inSpecs b) (outSpecs a `mappend` outSpecs b)
+instance Monad m => Semigroup (MkListeners m) where
+    a <> b = MkListeners act (inSpecs a <> inSpecs b) (outSpecs a <> outSpecs b)
       where
         act vI pD = (++) (mkListeners a vI pD) (mkListeners b vI pD)
 
+instance Monad m => Monoid (MkListeners m) where
+    mempty = MkListeners (\_ _ -> []) mempty mempty
```

Comment:
`(\_ _ -> [])` is just `mempty`. But some people might find lambda function more readable.